js/piwik.js

```diff
@@ -5584,6 +5590,58 @@ if (typeof window.Piwik !== 'object') {
 
                 return hookObj;
             }
+
+            var requestQueue = {
+                requests: [],
+                timeout: null,
+                sendRequests: function () {
+                    var requestsToTrack = this.requests;
+                    this.requests = [];
+                    if (requestsToTrack.length === 1) {
+                        sendRequest(requestsToTrack[0]);
+                    } else {
+                        sendBulkRequest(requestsToTrack);
+                    }
```

Comment:
Does sending a single request vs bulk request w/ single entry make a big difference?

Reply:
shouldn't make too much of a big difference. Although... for bulk requests we currently don't fallback to GET requests and always do a POST request compared to usually GET request, and it will be wrapped with `{"requests":["?' + requests.join('","?') + '"]}` so not sure if it can be replayed easily using log analytics etc.

I would say it is better when possible to use `sendRequest` when there is only one request and keep differentiating.

Reply:
I see 👍